### PR TITLE
Import metadata from commonly used keys

### PR DESCRIPTION
**Pull request by lhietal**
Read datetime, description and creator from commonly used metadata keys. [(IPTC Photo Metadata Mapping Guidelines)](https://iptc.org/std/photometadata/documentation/mappingguidelines/#the-mappings)

My original use case was to get datetime for images that didn't have Exif information. I added a few more keys that came up when reading the IPTC guide. The most significant change is in reading _creator_ value from IPTC IIM. _By-line_ is preferred over _Writer/Editor_. This is what I would expect based on the description of the keys.

- _2:122 Writer/Editor_, Identification of the name of the person involved in the writing,
editing or correcting the objectdata or caption/abstract.
- _2:80 By-line_, Contains name of the creator of the objectdata, e.g. writer, pho-
tographer or graphic artist. [(IPTC - NAA INFORMATION INTERCHANGE MODEL)](https://iptc.org/std/IIM/4.2/specification/IIMV4.2.pdf)

Time zone information in IIM _TimeCreated_ is just removed. This is not good but dt doesn't handle time zone at the moment so this is a separate issue.

### CODE REVIEW

File: src/common/exif.cc
Comment (kmilos) [inline src/common/exif.cc:1202]: Not sure these are the same thing: https://exiftool.org/forum/index.php?topic=10029.msg52187#msg52187

Also, did you test this? The `Exiv2::CommentValue` is made to specifically handle the 8 bytes of mandatory comment coding prefix and to decode encoded payload of `UserComment`, haven't checked what happens if you feed an ordinary ASCII string of `ImageDescription` to it...

Might be an idea to handle these separately?

Comment (lhietal) [reply src/common/exif.cc:1202]: I didn't find any problems with the current implementation but I agree that it would be better to handle them separately. I'll change that.

I added the _ImageDescription_ because it is similar to _dc:description_ [(IPTC Photo Metadata Mapping Guidelines)](https://iptc.org/std/photometadata/documentation/mappingguidelines/#description). I have no opinion if _UserComment_ should also go into description.

Comment (lhietal) [reply src/common/exif.cc]: Thanks for reviewing.

**Comment by TurboGit** [comment]
@lhietal : Seems like I completely missed this. I suppose it is still ok, right? I'll plan for 4.4. BTW, do not hesitate to ping me if you see that some PR seems ignored.

**Comment by lhietal** [comment]
@TurboGit: Thank you for reminding me about this PR. Yes this is still ok to merge.

**Comment by TurboGit** [comment]
@lhietal : I think it is ready to merge and I'll merge as soon as you provide a little text for the release notes. TIA. Sorry for the huge delay, this has by 2 times passed under the radar...

**Review by TurboGit — APPROVED** [review]
Ready to merge, waiting for the release notes and I'll push the button... promise :)

**Comment by lhietal** [comment]
@TurboGit Thanks for the review. Release notes are split in two because the first could affect someone's workflow and the other might be a bug fix.

Release notes:
Change reading _creator_ metadata from IPTC IIM to prefer _By-line_ over _Writer/Editor_.
Read _date/time_ and _description_ metadata from commonly used properties.

**Comment by TurboGit** [comment]
Thanks, testing on master I see some errors making the build failing. I'll fix (as I should have been reviewing this before :) and will merge another branch/PR. Closing this one. Thanks.

**Pull request closed without merging**
